### ISSUE TITLE
Testing for interaction effects (three-way interaction) running into unexpected missing values

### ISSUE THREAD

**Issue by weikang9009** [vincentarelbundock/marginaleffects#1161]
Hello! Thank you for developing and maintaining such an amazing package!

I tried to use your package to conduct testing of Average Marginal Effects (AMEs) by subgroups (essentially three-way interactions among three-factor variables) after estimating a logistic regression model with `survey.` 
```R
design_EI1=svydesign(ids=~1, weights=~PWEIGHT, data=data_match)
m2 = svyglm(EI~RaceEthnicity*Gender*Income+Age+I(Age^2)+MS+Size+
              Children+Tenure+Educ+Week, 
            design=design_EI1, family=binomial())
```
The testing seems to work if testing for gender gap among racial and income subgroups:
```R
ame_RaceEthnicity = avg_slopes(m2, variables = "Gender", by = c("RaceEthnicity","Income"), 
           newdata= data_match, wts = "PWEIGHT", 
           hypothesis =~reference)
```
Results are like this without any missing values
<img width="1321" alt="image" src="https://github.com/vincentarelbundock/marginaleffects/assets/7359284/74bbb641-6a68-4e46-8c7d-0c38e4b68f88">

However, if I switch the variables between the arguments "variables" and "by", I ran into many missing values:
```R
ame_Income = avg_slopes(m2, variables = "Income", by = c("RaceEthnicity","Gender"), 
           newdata= data_match,wts = "PWEIGHT", 
           hypothesis =~reference)
```
<img width="1320" alt="image" src="https://github.com/vincentarelbundock/marginaleffects/assets/7359284/dbe54141-17cc-41f4-9596-84be51b61235">

<img width="1332" alt="image" src="https://github.com/vincentarelbundock/marginaleffects/assets/7359284/d9d55ae2-c445-463b-a791-4df47e622eab">

Similar missing value and incomplete results if I set `variable="RaceEthnicity"`

Any insights will be greatly appreciated. Thank you!

**Comment by vincentarelbundock** [comment]
Interesting. I'm not sure. If you supply a working replicable and minimal example, I can look into it more deeply. Here are some tips on how to craft such an example: https://marginaleffects.com/vignettes/help.html#getting-help

**Comment by weikang9009** [comment]
Thanks for the quick response and offering to help! I am using the SmokeBan data as an example (the regression specification might not make sense theoretically - just using this to replicate the issue I ran into):
```R
library("marginaleffects")
library("survey")
dat <- "https://vincentarelbundock.github.io/Rdatasets/csv/AER/SmokeBan.csv"
dat <- read.csv(dat, na.strings = c("*", ""))
dat$weights <- runif(n=nrow(dat), min=1, max=100)
dat$smoker <-  fct_recode(dat$smoker, "0"= "no", "1" = "yes")
design1=svydesign(ids=~1, weights=~weights, data=dat)
m <- svyglm(smoker ~ ban*education*gender+age, design=design1, family=binomial())
```
If I assign `ban` (2 levels) to `variables`, the returned results make sense and do not have NAs:

```R
ame_int = avg_slopes(m, variables = "ban", by = c("education","gender"), 
           newdata= dat,wts = "weights", 
           hypothesis =~reference)
print(ame_int)
```
If I assign `education` (more than 3 levels) to `variables`, the returned results have many NAs:
```R
ame_int = avg_slopes(m, variables = "education", by = c("ban","gender"), 
           newdata= dat,wts = "weights", 
           hypothesis =~reference)
print(ame_int)
```
<img width="1090" alt="image" src="https://github.com/vincentarelbundock/marginaleffects/assets/7359284/0125139d-eb74-45d4-84ce-a48e6d62a380">

**Comment by weikang9009** [comment]
BTW, the NA issue seems to only apply to the `svyglm` model object. A `glm` model object will result in the NAs.

**Comment by weikang9009** [comment]
Hey @vincentarelbundock! Any thoughts on this? Thanks!

**Comment by vincentarelbundock** [comment]
Thanks for the report. I think you found a bug in `survey` support. 

It should be fixed in version 0.21.0.4, available from Github.
